config/foreman-hiera.yaml

```diff
@@ -6,11 +6,14 @@ defaults:
 hierarchy:
   - name: "Custom user answers"
     path: "custom.yaml"
+  - name: "Scenario OS overrides"
+    path: "scenario/%{facts.kafo.scenario.id}/family/%{facts.os.family}-%{facts.os.release.major}.yaml"
+  - name: "Scenario overrides"
+    path: "scenario/%{facts.kafo.scenario.id}.yaml"
```

Comment:
Scenarios should not override answers. It's a bad experience to have exposed parameters that you're not able to use. I think it's time to redesign the classes instead.

I've been thinking for a while about to define `class { 'foreman_proxy::plugin::plugin': ... }` inside the foreman-proxy-content class and not expose it in the installer.

Reply:
For a user, it will seem awkward if they are looking hard enough to try to configure something and it not have affect. Arguably though, there are plenty of parameters that exist today that a user *should not* touch. Whether they are things that will just break, or they are integral to the scenario itself. We have no good answers to that. 

Given this module already includes `foreman_proxy::plugin::pulp` (https://github.com/theforeman/puppet-foreman_proxy_content/blob/master/manifests/init.pp#L176) couldn't we then drop it from the answers and let hiera do the declarations for us per scenario?

Reply:
> Given this module already includes `foreman_proxy::plugin::pulp` (https://github.com/theforeman/puppet-foreman_proxy_content/blob/master/manifests/init.pp#L176) couldn't we then drop it from the answers and let hiera do the declarations for us per scenario?

So that is something I have been thinking about and was roughly referring to. I was thinking about exposing the options to `foreman_proxy_content` but in practice it's very unlikely someone modifies this and if they do, they can use `custom-hiera.yaml` as well. I think that leads to a cleaner setup, both from a code as well as a user perspective.